javascript/commons/Countdown.js

```diff
@@ -106,75 +105,13 @@ liquipedia.countdown = {
-		}
-		if ( timerObjectNode.dataset.streamFacebook ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/facebook/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamFacebook ) + '"><i class="lp-icon lp-icon-21 lp-facebook"></i></a>' );
-		}
-		if ( timerObjectNode.dataset.streamHuomao ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/huomao/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamHuomao ) + '"><i class="lp-icon lp-icon-21 lp-huomaotv"></i></a>' );
-		}
-		if ( timerObjectNode.dataset.streamHuya ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/huya/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamHuya ) + '"><i class="lp-icon lp-icon-21 lp-huyatv"></i></a>' );
-		}
-		if ( timerObjectNode.dataset.streamKick ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/kick/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamKick ) + '"><i class="lp-icon lp-icon-21 lp-kick"></i></a>' );
-		}
-		if ( timerObjectNode.dataset.streamLoco ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/loco/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamLoco ) + '"><i class="lp-icon lp-icon-21 lp-loco"></i></a>' );
-		}
-		if ( timerObjectNode.dataset.streamMildom ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/mildom/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamMildom ) + '"><i class="lp-icon lp-icon-21 lp-mildom"></i></a>' );
-		}
-		if ( timerObjectNode.dataset.streamNimo ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/nimo/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamNimo ) + '"><i class="lp-icon lp-icon-21 lp-nimotv"></i></a>' );
-		}
-		if ( timerObjectNode.dataset.streamTrovo ) {
-			streamsarr.push( '<a href="' + mw.config.get( 'wgScriptPath' ) + '/Special:Stream/trovo/' + liquipedia.countdown.getStreamName( timerObjectNode.dataset.streamTrovo ) + '"><i class="lp-icon lp-icon-21 lp-trovo"></i></a>' );
-		}
-		if ( timerObjectNode.dataset.streamTl ) {
-			streamsarr.push( '<a href="https://tl.net/video/streams/' + timerObjectNode.dataset.streamTl + '" target="_blank"><i class="lp-icon lp-icon-21 lp-stream"></i></a>' );
-		}
 		let html = '<span class="timer-object-countdown-time">' + datestr + '</span>';
-		if ( datestr.length > 0 && streamsarr.length > 0 ) {
+		if ( datestr.length > 0 && timerObjectNode.dataset.streams.length > 0 ) {
 			html += ' - ';
-		}
-		if ( timerObjectNode.dataset.finished !== 'finished' ) {
-			html += streamsarr.join( ' ' );
+			html += timerObjectNode.dataset.streams;
 		}
```

Comment:
Do we even need a countdown to be responsible for inserting the HTML for the streams? Can't it just be it is generated on the Lua side since I consider both of those logics separate?

Reply:
stream stuff now completely outside of js